src/NuGet.Core/NuGet.Protocol/Model/LocalPackageSearchMetadata.cs

```diff
@@ -145,5 +147,28 @@ public string LoadFileAsText(string path)
             }
             return fileContent;
         }
+
+        /// <summary>
+        /// Points to an Icon, either Embedded Icon or IconUrl
+        /// </summary>
+        public Uri GetIcon()
```

Comment:
```suggestion
        public Uri GetIconUri()
```

A method named `GetIcon` sounds like it should be getting pixel data, not just a URL.

Reply:
Renamed!